core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -913,6 +936,11 @@ public int compareTo(RunningSplitInfo o)
                     .compare(threadId, o.getThreadId())
                     .result();
         }
+
+        private void interruptStuckSplitTasksIfRunning(Consumer<TaskId> interruptStuckSplitTasksCallbackFunction)
```

Comment:
we can just call this 

```
private void applyFunctionToTaskRunner(Consumer<TaskId> callbackFunction);
```
and the one in TaskRunner 

```
private synchronized void apply(RunningSplitInfo splitInfo, Consumer<TaskId> callbackFunction)
        {
            if (splitInfo != null && currentThreadSplitInfo == splitInfo) {
                callbackFunction.accept(stuckSplitInfo.getTaskId());
            }
        }
```

as the methods are generic. but would leave it upto you.

Reply:
Would like to keep using the current explicit name, since can't find a second use case that need the thread safe apply.

What's more, if we want to change name, there is more works to do, we need to switch to generic types instead of taskId.